helm/charts/hpe-csi-driver/templates/hpe-csi-controller.yaml

```diff
@@ -61,24 +61,27 @@ spec:
           volumeMounts:
             - name: socket-dir
               mountPath: /var/lib/csi/sockets/pluginproxy
+        {{- if semverCompare ">=1.17.0" .Capabilities.KubeVersion.GitVersion }}
         - name: csi-snapshotter
-          {{- if semverCompare "~1.12.0" .Capabilities.KubeVersion.GitVersion }}
-          image: {{ .Values.csiSnapshotterImage}}: {{- .Values.csiSnapshotterTagv0}}
-          {{- else if semverCompare "~1.13.0" .Capabilities.KubeVersion.GitVersion }}
-          image: {{ .Values.csiSnapshotterImage}}: {{- .Values.csiSnapshotterTagv1}}
-          {{- else }}
-          image: {{ .Values.csiSnapshotterImage}}: {{- .Values.csiSnapshotterTagv2}}
-          {{- end }}
+          image: quay.io/k8scsi/csi-snapshotter:v2.0.0-rc2
           args:
-            - "--csi-address=$(ADDRESS)"
             - "--v=5"
+            - "--leader-election=false"
+            - "--csi-address=$(ADDRESS)"
           env:
             - name: ADDRESS
               value: /var/lib/csi/sockets/pluginproxy/csi.sock
-          imagePullPolicy: {{ .Values.imagePullPolicy | quote }}
+          imagePullPolicy: "Always"
           volumeMounts:
             - name: socket-dir
-              mountPath: /var/lib/csi/sockets/pluginproxy
+              mountPath: /var/lib/csi/sockets/pluginproxy/
+        - name: snapshot-controller
+          image: quay.io/k8scsi/snapshot-controller:v2.0.0-rc2
+          args:
+            - "--v=5"
+            - "--leader-election=false"
+          imagePullPolicy: Always
```

Comment:
Why is this container included?  Based on my understanding of the design, only the sidecar is required.

Reply:
```The CSI external-snapshotter sidecar is split into two controllers, a common snapshot controller and a CSI external-snapshotter sidecar.
 ```
From the release page of k8s 1.17 https://kubernetes.io/blog/2019/12/09/kubernetes-1-17-feature-cis-volume-snapshot-beta/

also the following commit from Xing https://github.com/kubernetes-csi/external-snapshotter/commit/6308420635f1c6f831e7c9d942d15161f13f2e7f

Reply:
Cool.  Thanks.